docs/readmes/proposals/p017_mme_migrate_to_c++.md

```diff
@@ -0,0 +1,134 @@
+---
+id: p017_mme_migrate_to_c++
+title: C++ Migration
+hide_title: true
+---
+
+
+# Proposal: C++ Migration
+
+Author(s): [Author Name, Co-Author Name]
+
+Last updated: 04/19/2021
+
+Discussion at
+[https://github.com/magma/magma/issues/4074](https://github.com/magma/magma/issues/4074).
+
+## Abstract
+
+This document describes the steps to migrate progressively, from C to C++, the MME core.
+
+## Background
+
+Initially and historically, the development of MME magma was done only in C.
+
+Many third party libraries and features that were integrated and or developed around the MME core are written in C++.
+
+We are now to the point where, for an easier and cleaner way of integrating new C++ libraries, for the ease of the maintenance of the code, a C to C++ migration seems unavoidable.
+
+## Proposal
+
+The proposal is constrained by a lack of performance of the current context serialization library (protobuf).
+ 
+Because the replacement libraries may have a big impact on the MME core code, this library migration will be studied first.
+
+
+### State Serialization
+The serialization performance of protobuf is insufficient, so a serialization library that allows better performance must be selected.
+
+The bad performances are on the global serialization process (translation, memory allocation, copy).
+
+The replacement libraries considered are "cap'n proto" and "Flat buffers”, they support C/C++ standart API, binary output format and do not need memory copy (zero copy).
+
+
+The states that have to be serialized are (To be detailled):
```

Comment:
Just this miss-spelling of detailed!

Reply:
Ok I will correct this very soon, thanks

Reply:
done